### PR TITLE
Add madjax_macropad keyboard

### PR DESCRIPTION
**Pull request by guidoism**
## Description

Add new keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/madjax_macropad/config.h
Comment (waffle87) [inline keyboards/madjax_macropad/config.h:1]: This file should be removed.

File: keyboards/madjax_macropad/info.json
Comment (waffle87) [inline keyboards/madjax_macropad/info.json:1]: This file should be renamed to `keyboard.json`.

File: keyboards/madjax_macropad/info.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "maintainer": "guidoism",
+    "bootloader": "rp2040",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/madjax_macropad/info.json:13]:
```suggestion
        "nkro": true,
        "encoder": true
```

File: keyboards/madjax_macropad/info.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["GP26", "GP27", "GP28", "GP29"],
+        "rows": ["GP6", "GP7", "GP0", "GP4", "GP3"]
+    },
```
Comment (waffle87) [inline keyboards/madjax_macropad/info.json:18]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "GP1", "pin_b": "GP2"}
        ]
    },
```

File: keyboards/madjax_macropad/info.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "processor": "RP2040",
+    "url": "https://github.com/guidoism/madjax_macropad",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
+    "layouts": {
+        "LAYOUT_madjax": {
```
Comment (waffle87) [inline keyboards/madjax_macropad/info.json:27]:
```suggestion
        "LAYOUT": {
```

File: keyboards/madjax_macropad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+     * │ 7 │ 8 │ 9 │ + │
+     * ├───┼───┼───┼───┤
+     * │ 4 │ 5 │ 6 │ % │
+     * ├───┼───┼───┼───┤
+     * │ 1 │ 2 │ 3 │ = │
+     * ├───┼───┼───┼───┤
+     * │ 0 │00 │ . │Ent│
+     * └───┴───┴───┴───┘
+     */
+    [0] = LAYOUT_madjax(
```
Comment (waffle87) [inline keyboards/madjax_macropad/keymaps/default/keymap.c:24]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/madjax_macropad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     * │Hom│ ↑ │PgU│ + │
+     * ├───┼───┼───┤───┤
+     * │ ← │   │ → │ % │
+     * ├───┼───┼───┤───┤
+     * │End│ ↓ │PgD│ = │
+     * ├───┼───┼───┤───┤
+     * │Ins│   │Del│Ent│
+     * └───┴───┴───┘───┘
+     */
+    [1] = LAYOUT_madjax(
```
Comment (waffle87) [inline keyboards/madjax_macropad/keymaps/default/keymap.c:45]: This layer is also not accessible.
```suggestion
    [1] = LAYOUT(
```

File: keyboards/madjax_macropad/readme.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# MadJax Macropad
+
+*A short description of the keyboard/project*
+
```
Comment (waffle87) [inline keyboards/madjax_macropad/readme.md:4]:
```suggestion
```

File: keyboards/madjax_macropad/rules.mk
Comment (waffle87) [inline keyboards/madjax_macropad/rules.mk:1]: This file should be removed. `ENCODER_ENABLE` should be moved to `keyboard.json` as suggested above, and `ENCODER_MAP_ENABLE` cannot be enabled at the keyboard level.

File: keyboards/madjax_macropad/readme.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+* Keyboard Maintainer: [Guido Bartolucci](https://github.com/guidoism)
+* Hardware Supported: Seeed Studio XIAO RP2040
+* Hardware Availability: https://github.com/guidoism/madjax_macropad
```
Comment (waffle87) [inline keyboards/madjax_macropad/readme.md:7]: Doesn't seem to go anywhere?

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/madjax_macropad/info.json
Comment (guidoism) [reply keyboards/madjax_macropad/info.json:1]: Done.

File: keyboards/madjax_macropad/info.json (hunk 2 above)
Comment (guidoism) [reply keyboards/madjax_macropad/info.json]: Done.

File: keyboards/madjax_macropad/info.json (hunk 3 above)
Comment (guidoism) [reply keyboards/madjax_macropad/info.json]: Done.

File: keyboards/madjax_macropad/keymaps/default/keymap.c (hunk 4 above)
Comment (guidoism) [reply keyboards/madjax_macropad/keymaps/default/keymap.c]: Done.

File: keyboards/madjax_macropad/keymaps/default/keymap.c (hunk 5 above)
Comment (guidoism) [reply keyboards/madjax_macropad/keymaps/default/keymap.c]: Done.

File: keyboards/madjax_macropad/readme.md (hunk 6 above)
Comment (guidoism) [reply keyboards/madjax_macropad/readme.md]: Done.

File: keyboards/madjax_macropad/readme.md (hunk 7 above)
Comment (guidoism) [reply keyboards/madjax_macropad/readme.md:5]: Created repo that I'm filling in.

File: keyboards/madjax_macropad/rules.mk
Comment (guidoism) [reply keyboards/madjax_macropad/rules.mk:1]: I can't get my encoder to work without the `ENCODER_ENABLE` in `rules.mk`. The configuration of the pins definitely works from the json but actually enabling it doesn't. All of the docs still reference the C config and the [reference_info_json](https://github.com/qmk/qmk_firmware/blob/master/docs/reference_info_json.md#encoder-encoder) doesn't mention anything different than what you suggested adding. What to do?

Comment (waffle87) [reply keyboards/madjax_macropad/rules.mk:1]: By cannot be enabled at the keyboard level, I mean it needs to be enabled at the keymap level, in `keyboards/<keyboard_name>/keymaps/default/rules.mk`.

Comment (waffle87) [inline keyboards/madjax_macropad/rules.mk:1]: This file still needs to be removed. Enabling encoders belongs in `keyboard.json` under `features.encoder`.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/madjax_macropad/keymaps/default/keymap.c (hunk 5 above)
Comment (waffle87) [reply keyboards/madjax_macropad/keymaps/default/keymap.c]: This suggestion has not been made.

File: keyboards/madjax_macropad/rules.mk
Comment (guidoism) [reply keyboards/madjax_macropad/rules.mk:1]: Roger. Done.

Comment (guidoism) [reply keyboards/madjax_macropad/rules.mk:1]: Done.

File: keyboards/madjax_macropad/keymaps/default/keymap.c (hunk 5 above)
Comment (guidoism) [reply keyboards/madjax_macropad/keymaps/default/keymap.c]: Should be now.

File: keyboards/madjax_macropad/keymaps/default/rules.mk [hunk 8]
```diff
@@ -0,0 +1 @@
+ENCODER_ENABLE = yes
```
Comment (waffle87) [inline keyboards/madjax_macropad/keymaps/default/rules.mk:1]: Still no. `ENCODER_ENABLE` belongs at the keyboard level. This enables the hardware encoder feature. At the keymap level (this file) you want to enable the encoder configuration subsystem you are using (`ENCODER_MAP_ENABLE`).

File: keyboards/madjax_macropad/keymaps/default/keymap.c (hunk 5 above)
Comment (waffle87) [reply keyboards/madjax_macropad/keymaps/default/keymap.c]: > This layer is also not accessible.

File: keyboards/madjax_macropad/keymaps/default/rules.mk (hunk 8 above)
Comment (guidoism) [reply keyboards/madjax_macropad/keymaps/default/rules.mk]: Sorry, the change to the json is a bit confusing. The docs all refer to the old way of doing stuff. Do y'all need help updating the docs? 

I think I got it this time.

File: keyboards/madjax_macropad/keymaps/default/keymap.c (hunk 5 above)
Comment (guidoism) [reply keyboards/madjax_macropad/keymaps/default/keymap.c]: Removed it.

**Review by waffle87 — APPROVED** [review]

File: keyboards/madjax_macropad/keyboard.json [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    "bootloader": "rp2040",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
```
Comment (fauxpark) [inline keyboards/madjax_macropad/keyboard.json:14]:
```suggestion
        "nkro": true
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/madjax_macropad/keyboard.json (hunk 9 above)
Comment (guidoism) [reply keyboards/madjax_macropad/keyboard.json]: Changed back to strict json as requested.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**